packages/wonder-blocks-badge/src/components/badge.tsx

```diff
@@ -1,5 +1,151 @@
+export type BadgeProps = AriaProps & {
+    /**
+     * The label to display in the badge.
+     */
+    label?: string;
+    /**
+     * The icon to display in the badge. It can be a PhosphorIcon, a custom svg,
+     * or `img` element.
+     * Considerations:
+     * - If the icon conveys meaning, set the alt text on the icon being used
+     * - If the icon is an `img` element, it may need width: 100% and height: 100%
+     *   to render properly in the badge.
+     */
+    icon?: React.ReactElement;
+    /**
+     * The id for the badge.
+     */
+    id?: string;
+    /**
+     * The test id for the badge.
+     */
+    testId?: string;
+    /**
+     * Custom styles for the elements in the Badge component.
+     * - `root`: Styles the root element
+     * - `icon`: Styles the icon element
+     */
+    styles?: {
+        root?: StyleType;
+        icon?: StyleType;
+    };
 };
+
+const StyledDiv = addStyle("div");
+const StyledSpan = addStyle("span");
+
+/**
+ * Badges are visual indicators used to display concise information, such as
+ * a status, label, or count.
+ */
+const Badge = React.forwardRef<HTMLDivElement, BadgeProps>(function Badge(
+    props: BadgeProps,
+    ref: React.ForwardedRef<HTMLDivElement>,
+) {
+    const {icon, label, id, testId, styles: stylesProp, ...otherProps} = props;
+    if (!label && !icon) {
+        return <React.Fragment />;
+    }
+    return (
+        <StyledDiv
+            id={id}
+            data-testid={testId}
+            ref={ref}
+            style={[
+                typographyStyles.LabelXSmall,
+                styles.badge,
+                styles.defaultBadgeStyling,
+                icon && !label ? styles.iconOnly : {},
+                stylesProp?.root,
+            ]}
+            {...otherProps}
+        >
+            {icon && (
+                <StyledSpan
+                    style={[
+                        styles.icon,
+                        styles.defaultIconStyling,
+                        stylesProp?.icon,
+                    ]}
+                >
+                    {icon}
+                </StyledSpan>
+            )}
+            {label}
+        </StyledDiv>
+    );
+});
+
+export {Badge};
+
+const badgeTokens = {
+    root: {
+        sizing: {
+            paddingBlock: sizing.size_040,
+            paddingInline: sizing.size_080,
+            gap: sizing.size_040,
+        },
+        border: {
+            width: border.width.thin,
+            style: "solid",
+            radius: border.radius.radius_080,
+        },
+        color: {
+            background: semanticColor.surface.secondary,
+            foreground: semanticColor.text.primary,
+            border: semanticColor.border.subtle,
+        },
+    },
+    icon: {
+        sizing: {
+            width: sizing.size_160,
+            height: sizing.size_160,
+        },
+        color: {
+            foreground: semanticColor.icon.primary,
+        },
+    },
+    iconOnly: {
```

Comment:
`iconOnly` styles... I'm not sure about where to place the padding token! For context, the badge has smaller padding when it is using only an icon. It's more of a variant of the `root` element, but I'm not sure how to structure it. I'm open to changing this!

Reply:
How do you feel about using the `layout` category? (described in the naming guide).

Reply:
Good call! I was mixing up `layout` and `sizing`. I update the grouping for padding/gap properties to `layout`!